data/serializer/src/main/java/tech/pegasys/teku/api/schema/merge/BeaconStateMerge.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api.schema.merge;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+import java.util.Objects;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.teku.api.schema.BeaconBlockHeader;
+import tech.pegasys.teku.api.schema.Checkpoint;
+import tech.pegasys.teku.api.schema.Eth1Data;
+import tech.pegasys.teku.api.schema.Fork;
+import tech.pegasys.teku.api.schema.Validator;
+import tech.pegasys.teku.api.schema.altair.BeaconStateAltair;
+import tech.pegasys.teku.api.schema.altair.SyncCommittee;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.MutableBeaconState;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.merge.BeaconStateSchemaMerge;
+import tech.pegasys.teku.ssz.SszList;
+import tech.pegasys.teku.ssz.collections.SszBitvector;
+import tech.pegasys.teku.ssz.primitive.SszByte;
+
+public class BeaconStateMerge extends BeaconStateAltair {
+  @JsonProperty("latest_execution_payload_header")
+  public final ExecutionPayloadHeader latestExecutionPayloadHeader;
+
+  public BeaconStateMerge(
```

Comment:
should have @JSonCreator at the start as the properties below are only set because of it being a creator...

the genesis_validators_root property doesnt look like its come through correctly, its coming up with 'empty' and 'zero' attributes in the swagger on the altair state, so that suggests it needs an annotation (`@Schema`)

Also the merge state isn't listed, so its missing from the `oneOf` in BeaconState.

Reply:
i think `genesis_validators_root` looks not good in master too, fixing :)
going to check the @JsonCreator
the missing `oneOf` is actually not missing... you might hadn't rebuild the latest version?